public/components/common/hooks/use-filter-manager.ts

```diff
@@ -10,10 +10,24 @@
  * Find more information about this on the LICENSE file.
  */
 import { getDataPlugin } from '../../../kibana-services';
-import { useState, useEffect} from 'react';
-
+import { useState, useEffect, useMemo } from 'react';
+import { Filter } from 'src/plugins/data/public';
+import _ from 'lodash';
 
 export const useFilterManager = () => {
-    const [filterManager, setFilterManager] = useState(getDataPlugin().query.filterManager);
-    return filterManager;
-}
\ No newline at end of file
+  const filterManager = useMemo(() => getDataPlugin().query.filterManager, []);
+  const [filters, setFilters] = useState<Filter[]>([])
+
+  useEffect(() => {
+    let filterSubscriber = filterManager.getUpdates$().subscribe(() => {
+      const newFilters = filterManager.getFilters();
+      if (!_.isEqual(filters, newFilters)) {
+        setFilters(newFilters);
+      }
+      return () => {
+        filterSubscriber.unsubscribe();
+      };
```

Comment:
**issue**: The effect function of `useEffect` should return a function that will be called when the component will unmount to unsubscribe.

**suggestion**: Fix the returned function
```ts
useEffect(() => {
  // Suggestion change: destructuring the `unsubscribe` method
  const { unsubscribe } = filterManager.getUpdates$().subscribe(() => {
    const newFilters = filterManager.getFilters();
    if (!_.isEqual(filters, newFilters)) {
      setFilters(newFilters);
    }
  }
  // The returned function that will be called when the component will unmount is returned from the effect function. In the current code, a function is returned from the subscription callback function.
  return unsubscribe
}

Reply:
It seems functionally the same but I do like the elegance of it. I'll commit it right away.